src/pudl/transform/eiaaeo.py

```diff
@@ -560,6 +514,115 @@ def core_eiaaeo__yearly_projected_generation_in_end_use_sectors_by_fuel_type(
     return renamed_for_pudl
 
 
+@asset(io_manager_key="pudl_io_manager")
+def core_eiaaeo__yearly_projected_fuel_cost_in_electric_sector_by_type(
+    raw_eiaaeo__electric_power_projections_regional,
+):
+    """Projected fuel cost for the electric power sector.
+
+    Includes 2022 US dollars per million BTU and nominal US dollars per million
+    BTU.
+
+    In future report years, the base year for the real cost will change, so we
+    store that base year as well.
+    """
+    sanitized = filter_enrich_sanitize(
+        raw_df=raw_eiaaeo__electric_power_projections_regional,
+        relevant_series_names=("Electricity : Fuel Prices",),
+    ).rename(columns={"subtopic": "variable_name", "variable_name": "dimension"})
+
+    assert set(sanitized.topic.unique()) == {"electricity"}
+    assert set(sanitized.variable_name.unique()) == {"fuel_prices"}
+    assert set(sanitized.units.unique()) == {"2022_mmbtu", "nom_mmbtu"}
+    # turn variable_name into `nominal_fuel_prices` and `real_fuel_prices` based on unit
+    sanitized.variable_name = sanitized.units + "_" + sanitized.variable_name
+
+    trimmed = sanitized.drop(
+        columns=[
+            "topic",
+            "units",
+        ]
+    )
+
+    unstacked = unstack(
+        df=trimmed,
+        eventual_pk=[
+            "report_year",
+            "model_case_eiaaeo",
+            "region",
+            "dimension",
+            "projection_year",
+        ],
+    ).assign(real_cost_basis_year=2022)
+
+    renamed_for_pudl = unstacked.reset_index().rename(
+        columns={
+            "capacity": "summer_capacity_mw",
+            "generation": "gross_generation_mwh",
+            "region": "electricity_market_module_region_eiaaeo",
+            "dimension": "fuel_type_eiaaeo",
+            "2022_mmbtu_fuel_prices": "fuel_cost_real_per_mmbtu_eiaaeo",
+            "nom_mmbtu_fuel_prices": "fuel_cost_per_mmbtu",
+        }
+    )
+    return renamed_for_pudl
+
+
+@dataclass
+class AeoCheckSpec:
+    """Define some simple checks that can run on any AEO asset."""
+
+    name: str
+    asset: str
+    num_rows_by_report_year: dict[int, int]
+    category_counts: dict[str, int]
+
+
+BASE_AEO_CATEGORIES = {
+    "model_case_eiaaeo": 17,
+    "projection_year": 30,
+    "electricity_market_module_region_eiaaeo": 26,
+}
+check_specs = [
+    AeoCheckSpec(
+        name="gen_in_electric_sector_by_tech",
+        asset="core_eiaaeo__yearly_projected_generation_in_electric_sector_by_technology",
+        num_rows_by_report_year={2023: 166972},
+        category_counts=BASE_AEO_CATEGORIES
+        | {
+            "technology_description_eiaaeo": 13,
```

Comment:
Is there a reason these aren't in test modules?

Reply:
These are for defining asset checks, which get run during the ETL after an asset has been materialized.